src/openforms/js/components/admin/form_design/logic/actions/dmn/DMNActionConfig.js

```diff
@@ -104,13 +107,15 @@ const DecisionDefinitionVersionField = () => {
     // If decisionDefinitionId has changed, reset the value of the definition version if the new decision definition
     // does not have that version
     if (
-      values.decisionDefinitionVersion !== '' &&
-      !versionChoices.find(item => item[0] === values.decisionDefinitionVersion)
+      decisionDefinitionVersion !== '' &&
```

Comment:
shouldn't `decisionDefinitionVersion` be passed as dependency to `useAsync` too?

Reply:
I thought no, because we don't need to refetch/update things if the version changes :thinking:

Reply:
but it's used inside the hook in a check? if the version changes, the callback still has the old value and behaviour is not as expected.

Looking at it again, I think the `setFieldValue` call based on the available version choices actually belongs in a separate `useEffect`. And you can also simplify the logic IMO - if the definition ID changes, then the version should always be reset to make users aware they explicitly need to select the version. Version X for definition A does not have the same semantic meaning as Version X for definition B.